### DISCUSSION TITLE
Looking for information on MachineId format

### DISCUSSION THREAD

**Discussion by LelouBil** [SteamRE/SteamKit#1165] [General]
Hello, I am trying to reproduce the MachineId generation of the steam client.

I have successfully reproduced the first field and from the last field using code from here : 
https://github.com/SteamRE/SteamKit/blob/8e7ec9461c88373611f5f089c2152e9277bfa03a/SteamKit2/SteamKit2/Util/HardwareUtils.cs#L45

However, I have not been able to reproduce the second field, based on the Mac address of the interface.
I even tried every interface on my system, but I am not able to get the same hash value as the one steam is sending for the second field.

To test it, I intercepted the login request using NetHook2 and NetHookAnalyzer2, and used a hex editor to look at the hashes.

**Comment by zxcvqwerasdf** [comment]
I think it's because steam machine id generation != steamkit machine id generation. You need reverse engineer machineid gathering from steam, no other ways.
